### PR TITLE
feat: add ability to sort languages by vitality (#215)

### LINKED ISSUE #215: Vitality: Add ability to sort languages by vitality
People may want to sort languages by their vitality.

* `PageParamTypes.tsx` is where we indicate the possible sorts
* `sort.tsx` is where we put the algorithms
  * Add a new case to the switch block in`getSortFunctionParameterized`.
  * Only objects where the `object.type = ObjectType.Language` can have the 0 to 9 vitality numbers -- so have it get a value of -1.
  * Convert the raw values to the scores -- just make sure to set the default sort direction to be high to low
  * Since vitality is only a thing for languages, you'll just add this vitality option to `getSortBysApplicableToObjectType` 
* If you've added the Vitality columns to the table, you can add these new values as the sortKeys

### PR DESCRIPTION
**Pull request by abdulkalam556**
## Summary
Adds sorting options for languages based on vitality values, along with unit and integration tests to verify the functionality.

## Changes
- **Extended `SortBy` enum** with:
  - Vitality: Metascore
  - Vitality: ISO
  - Vitality: Ethnologue 2013
  - Vitality: Ethnologue 2025
- **Implemented sort functions** in `sort.tsx`:
  - Type-cast `ObjectData` → `LanguageData`
  - Convert string vitality values to numeric via `parseInt`
  - Metascore derived as the average of available vitality values (with fallback logic)
- **Added vitality sort keys** to `getSortBysApplicableToObjectType` for `ObjectType.Language`.
- **Added unit tests** in `/src/views/language/test/LanguageVitalityComputation.test.tsx`:
  - Verify scoring functions convert raw string values into normalized 0–9 scores
  - Check edge cases: unknown values, empty strings, nulls
  - Validate metascore priority and fallback logic
- **Added integration tests** in `/src/controls/test/sort.test.tsx`:
  - Ensure languages with higher vitality scores sort before lower ones
  - Verify non-language objects always sort last
  - Confirm missing vitality values are handled correctly
  - Test all four vitality sort cases (Metascore, ISO, Ethnologue 2013, Ethnologue 2025)

## Notes
- Non-language objects return `-1` when sorted by vitality and are pushed to the end.
- Default sort order is **descending** (high → low).
- Tests included to prevent regressions and validate expected behavior.
- Closes #215.

## Checklist
- [x] Code changes implemented
- [x] Unit tests created for scoring functions
- [x] Integration tests created for sort functions
- [x] Verified default sort order is high → low
- [x] PR linked to issue #215

### CODE REVIEW

File: src/controls/sort.tsx [hunk 1]
```diff
@@ -227,3 +251,13 @@ function getCountOfTerritories(object: ObjectData): number {
...
+
+function makeVitalityComparator(
+  getScore: (lang: LanguageData) => number | null | undefined
+): SortByFunctionType {
+  return (a: ObjectData, b: ObjectData) => {
+    const aVal = a.type === ObjectType.Language ? getScore(a) ?? -1 : -1;
+    const bVal = b.type === ObjectType.Language ? getScore(b) ?? -1 : -1;
+    return bVal - aVal; // higher score first
+  };
+}
```
Comment (conradarcturus) [inline src/controls/sort.tsx]: This looks great! Thanks. For better or worse I implemented a generic version of this -- more complicated but handles more cases. You'll want to use this now:

https://github.com/Translation-Commons/lang-nav/blob/master/src/controls/sort.tsx#L71-L86

**Review by conradarcturus — APPROVED** [review]
almost there! just fix the merge conflict
